lib/pool.dart

```diff
@@ -152,26 +152,26 @@ class Pool {
   /// to, a [StateError] is thrown.
   Stream<T> forEach<S, T>(
       Iterable<S> elements, FutureOr<T> Function(S source) action,
-      {bool Function(S item, Object error, StackTrace stack) onError}) {
+      {bool Function(S item, Object error, StackTrace stack)? onError}) {
     onError ??= (item, e, s) => true;
 
     var cancelPending = false;
 
-    Completer resumeCompleter;
-    StreamController<T> controller;
+    Completer? resumeCompleter;
+    late StreamController<T> controller;
 
-    Iterator<S> iterator;
+    Iterator<S>? iterator;
 
     Future<void> run(int i) async {
-      while (iterator.moveNext()) {
+      while (iterator!.moveNext()) {
```

Comment:
nit: assign to a local variable to avoid the repeated `!`

Reply:
https://github.com/dart-lang/pool/pull/36